cm-server/src/main/java/ro/msg/cm/controller/UpdateStudyYearController.java

```diff
@@ -0,0 +1,178 @@
+package ro.msg.cm.controller;
+
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.hateoas.Resource;
+import org.springframework.hateoas.Resources;
+import org.springframework.web.bind.annotation.PathVariable;
+import org.springframework.web.bind.annotation.RequestMapping;
+import org.springframework.web.bind.annotation.RequestMethod;
+import org.springframework.web.bind.annotation.RestController;
+import ro.msg.cm.model.Candidate;
+import ro.msg.cm.model.StartYearProperties;
+import ro.msg.cm.repository.CandidateRepository;
+
+import java.text.ParseException;
+import java.text.SimpleDateFormat;
+import java.util.*;
+
+import static java.util.Calendar.YEAR;
+import static org.springframework.hateoas.mvc.ControllerLinkBuilder.linkTo;
+import static org.springframework.hateoas.mvc.ControllerLinkBuilder.methodOn;
+
+@RestController
+@RequestMapping("/api")
+public class UpdateStudyYearController {
+    private final CandidateRepository candidateRepository;
+    private final StartYearProperties startYearProperties;
+
+    @Autowired
+    public UpdateStudyYearController(CandidateRepository candidateRepository, StartYearProperties startYearProperties) {
+        this.candidateRepository = candidateRepository;
+        this.startYearProperties = startYearProperties;
+    }
+
+    @RequestMapping(value = "/candidates", method = RequestMethod.GET, produces = "application/hal+json")
+    public Resources updateUniversityYear() throws ParseException {
+
+        Iterable<Candidate> candidates = findAll();
```

Comment:
I would remove the findAll method and simply use the repository method (basically the findAll anyway just wraps the repository itself).